travel-planner/travel-planner-ui/src/Components/Maps/Maps.jsx

```diff
@@ -37,44 +33,91 @@ export default function Maps({
 
 		// get the markers
 		useEffect(async () => {
-			const res = await axios.post(`http://localhost:${PORT}/maps/getMarkers`, {
-				mapId: trip,
-			});
-			setMarkers(res.data);
-		}, []);
-
-		if (!markers) {
-			return <Loading />;
-		}
+			try {
+				const res = await axios.post(
+					`http://localhost:${PORT}/maps/getMarkers`,
+					{
+						mapId: trip.objectId,
+					}
+				);
+				setMarkers(res.data.markers);
+				setUpdate(false);
+			} catch (error) {
+				console.log("Error fetching markers: ", error);
```

Comment:
Probably consider follow up adding error UI. It can be a common error UI for all abnormal cases.